pkg/sdn/plugin/cniserver.go

```diff
@@ -0,0 +1,213 @@
+package plugin
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/gorilla/mux"
+)
+
+const CniServerSocketPath string = "/var/run/openshift-sdn/cni-server.sock"
+
+type CniCommand string
+
+const CNI_ADD CniCommand = "ADD"
+const CNI_UPDATE CniCommand = "UPDATE"
```

Comment:
It's mostly to ensure requests are serialized; eg that an UPDATE cannot happen concurrently with an ADD.  Does that make more sense?